python/ccf/ledger.py

```diff
@@ -493,6 +567,20 @@ def __next__(self):
         return self
 
 
+class Snapshot(Entry):
+    """
+    Utility used to parse the content of a snapshot file.
+    """
+
+    def __init__(self, filename: str):
+        super().__init__(filename)
+        self._file_size = os.path.getsize(filename)
+        super()._read_header()
+
+    def __del__(self):
```

Comment:
Not guaranteed to execute, not a good way to do cleanup (or anything in fact).

Reply:
Good point. That had been there forever but is now fixed.